modules/seqexec/server/src/main/scala/seqexec/server/gnirs/GnirsEpics.scala

```diff
@@ -126,100 +133,99 @@ class GnirsEpics(epicsService: CaService, tops: Map[String, String]) {
+  private val dcState: CaStatusAcceptor = epicsService.getStatusAcceptor("nirs::dcstatus")
 
-  def arrayType: Option[String] = Option(dcState.getStringAttribute("arraytyp").value)
+  def arrayId: F[Option[String]] = safeAttribute(dcState.getStringAttribute("arrayid"))
 
-  def obsEpoch: Option[Double] = Option(dcState.getDoubleAttribute("OBSEPOCH").value).map(_.toDouble)
+  def arrayType: F[Option[String]] = safeAttribute(dcState.getStringAttribute("arraytyp"))
 
-  def detBias: Option[Double] = Option(dcState.getDoubleAttribute("detBias").value).map(_.toDouble)
+  def obsEpoch: F[Option[Double]] = Nested(safeAttributeSDouble(dcState.getDoubleAttribute("OBSEPOCH"))).map(_.toDouble).value
 
-  def countDown: Option[String] = Option(dcState.getStringAttribute("countdown").value)
+  def detBias: F[Option[Double]] = Nested(safeAttributeSDouble(dcState.getDoubleAttribute("detBias"))).map(_.toDouble).value
 
-  def numCoadds: Option[Int] = Option(dcState.getIntegerAttribute("numCoAdds").value).map(_.toInt)
+  def countDown: F[Option[String]] = safeAttribute(dcState.getStringAttribute("countdown"))
 
-  def wcs: Option[String] = Option(dcState.getStringAttribute("wcs").value)
+  def numCoadds: F[Option[Int]] = Nested(safeAttributeSInt(dcState.getIntegerAttribute("numCoAdds"))).map(_.toInt).value
 
-  def exposureTime: Option[Double] = Option(dcState.getDoubleAttribute("exposureTime").value).map(_.toDouble)
+  def wcs: F[Option[String]] = safeAttribute(dcState.getStringAttribute("wcs"))
 
-  def digitalAvgs: Option[Int] = Option(dcState.getIntegerAttribute("digitalAvgs").value).map(_.toInt)
+  def exposureTime: F[Option[Double]] = Nested(safeAttributeSDouble(dcState.getDoubleAttribute("exposureTime"))).map(_.toDouble).value
 
-  def lowNoise: Option[Int] = Option(dcState.getIntegerAttribute("lowNoise").value).map(_.toInt)
+  def digitalAvgs: F[Option[Int]] = Nested(safeAttributeSInt(dcState.getIntegerAttribute("digitalAvgs"))).map(_.toInt).value
 
-  private val observeCAttr: CaAttribute[CarStateGEM5] = dcState.addEnum("observeState",
-    s"${GnirsTop}dc:observeC.VAL", classOf[CarStateGEM5])
-  def observeState: Option[CarStateGEM5] = Option(observeCAttr.value)
+  def lowNoise: F[Option[Int]] = Nested(safeAttributeSInt(dcState.getIntegerAttribute("lowNoise"))).map(_.toInt).value
 
-  def dhsConnected: Option[Boolean] = Option(dcState.getIntegerAttribute("dhsConnected").value)
-    .map(_.toInt =!= 0)
+  def dhsConnected: F[Option[Boolean]] = Nested(safeAttributeSInt(dcState.getIntegerAttribute("dhsConnected")))
+    .map(_.toInt =!= 0).value
 
   val arrayActiveAttr: Option[CaAttribute[JDetectorState]] = Option(dcState.addEnum(
     "arrayState", s"${GnirsTop}dc:activate", classOf[JDetectorState]
   ))
-  def arrayActive: Option[Boolean] = arrayActiveAttr.flatMap(at => Option(at.value))
-    .map(_.getActive)
 
-  def minInt: Option[Double] = Option(dcState.getDoubleAttribute("minInt").value).map(_.toDouble)
+  def arrayActive: F[Option[Boolean]] =
```

Comment:
@jluhrs Could you review this one in particular. I'm not sure if I fully understand the original

Reply:
The original takes the attribute, which reads a binary enum, and converts it to a `Boolean`. Of course, the attribute could not exists, that is why it is treated a an Option. Your implementation follows that, but I don't understand the last `_.getOrElse(false)`. There is no default value in the original. If the value cannot be read, then the whole result is `None`. That us why there is a `flatMap`.